### PR TITLE
Move examples from cog-examples to cog

### PR DESCRIPTION
**Pull request by anish-sahoo**
Moves the example models from the separate `replicate/cog-examples` repo into this repo under `examples/`, migrates them to the `BaseRunner`/`run` API, and adds `cog.Secret` input support to coglet.

- Add example models using `run.py`/`BaseRunner`: `blur`, `canary`, `hello-concurrency`, `hello-context`, `hello-image`, `hello-replicate`, `hello-train`, `hello-world`, `notebook`, `z-image-turbo`
- Migrate the existing `streaming-text` example to `run.py`/`BaseRunner`
- Replace `examples/resnet` with the classic torchvision ResNet50 classifier (`BaseRunner`, ImageNet weights bundled with torchvision -- no managed weights or import step), ported and modernized from `replicate/cog-examples`
- Move the previous managed-weights ResNet example to `examples/experimental/resnet-managed-weights` (renamed `model: resnet-managed-weights`)
- Update `managed-weights` `cog.yaml` to use the `model:` field (replacing `image: <your-registry>/...`) and refresh its `weights.lock`
- `hello-replicate` demonstrates `Secret` usage by calling the Replicate API

### coglet
- Add `cog.Secret` input support: classify and coerce `Secret`, `Optional[Secret]`, and `Secret | None` fields, wrapping string values in `cog.types.Secret` (`list[Secret]` is intentionally not coerced)
- Ignore two pyo3 0.27 advisories in `cargo-deny` (`RUSTSEC-2026-0176`, `RUSTSEC-2026-0177`) with justification; pinned due to numpy/pyo3-async-runtimes lag, and neither code path is reachable in coglet

### test harness
- Add a `base_dir` field for `repo: local` models so local models resolve relative to the manifest (defaults to `fixtures/models/`)
- Point all example models in `manifest.yaml` at the local `examples/` directory instead of `replicate/cog-examples`
- Migrate fixture models from `predict.py`/`BasePredictor` to `run.py`/`BaseRunner`
- Only pass `--setup-timeout` to `cog predict` (`cog train` does not support it)

### CODE REVIEW

File: examples/managed-weights/weights.lock [hunk 1]
```diff
@@ -5,93 +5,91 @@
...
+          "digest": "sha256:b141de6ec6d7f982ece13f98f604e3fe1807ea9c0e839185d0ab7064604209d0",
+          "layer": "sha256:f7269ec7efd927918d3c565d045f47bfcbf778c8f14bed3d8630d7cb5026c0da"
         },
         {
           "path": "model.safetensors",
-          "size": 2508311120,
-          "digest": "sha256:3a2026366188c8c68598edbbff92f8d11590a08e0ae2e6775544e7b07d6a5e11",
-          "layer": "sha256:65f11b0713429c604eb159d02a1c805815d599414e10c1c295e1937472db3a2f"
+          "size": 135,
+          "digest": "sha256:960712aba3c75945099daa9a711b466a05c0b9952be05596d764e2f329174e12",
```
[the remark this answers is not in the document]
Comment (anish-sahoo) [reply examples/managed-weights/weights.lock]: Regenerated weights.lock from the real ~2.5GB parakeet weights (cloned via git-lfs and re-ran `cog weights import`). model.safetensors and the parakeet total now report correct sizes/digests.

File: examples/managed-weights/weights.lock [hunk 2]
```diff
@@ -5,93 +5,91 @@
...
+          "layer": "sha256:f7269ec7efd927918d3c565d045f47bfcbf778c8f14bed3d8630d7cb5026c0da"
         },
         {
```
[the remark this answers is not in the document]
Comment (anish-sahoo) [reply examples/managed-weights/weights.lock:43]: Regenerated weights.lock from the real ~2.5GB parakeet weights (cloned via git-lfs and re-ran `cog weights import`). model.safetensors and the parakeet total now report correct sizes/digests.

File: examples/z-image-turbo/run.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    def run(self, prompt: str) -> Path:
+        image = self.model(
+            prompt=prompt,
+            height=1024,
+            width=1024,
+            num_inference_steps=9,  # This actually results in 8 DiT forwards
+            guidance_scale=0.0,  # Guidance should be 0 for the Turbo models
+            generator=torch.Generator("cuda").manual_seed(42),
+        ).images[0]
+        output_path = Path(tempfile.mktemp(suffix=".png"))
```
[the remark this answers is not in the document]
Comment (anish-sahoo) [reply examples/z-image-turbo/run.py]: Fixed: replaced tempfile.mktemp() with tempfile.NamedTemporaryFile(suffix='.png', delete=False).

File: examples/hello-concurrency/run.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    format="%(asctime)s %(levelname)-8s %(message)s",
+    level=logging.INFO,
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+
+honeycomb_token = ""
+try:
+    with open("./honeycomb_token.key", "r") as f:
+        honeycomb_token = f.read().strip()
+except FileNotFoundError:
```
[the remark this answers is not in the document]
Comment (anish-sahoo) [reply examples/hello-concurrency/run.py:35]: Fixed: replaced the empty except with a logging.info(...) documenting that OTEL is disabled when the token file is absent.

File: examples/blur/run.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+
+class Runner(BaseRunner):
+    def run(
+        self,
+        image: Path = Input(description="Input image"),
+        blur: float = Input(description="Blur radius", default=5),
+    ) -> Path:
+        if blur == 0:
+            return input
```
[the remark this answers is not in the document]
Comment (anish-sahoo) [reply examples/blur/run.py]: Fixed: now returns image instead of the built-in input.

File: tools/test-harness/manifest.yaml [hunk 6]
```diff
@@ -113,18 +123,10 @@ models:
...
-    # NOTE: `cog train` in the RC may have input validation issues
-    # (validates against predict schema instead of train schema).
-    # The train_test below may fail — that's a real compatibility signal.
-    train_tests:
-      - description: "train produces weights file"
-        inputs:
-          prefix: "custom"
-        expect:
-          type: not_empty
     tests:
```
[the remark this answers is not in the document]
Comment (anish-sahoo) [reply tools/test-harness/manifest.yaml:133]: this is deprecated so not needed

[the remark this answers is not in the document]
Comment (anish-sahoo) [reply tools/test-harness/manifest.yaml:133]: depracted, we have replacement code with same functionality

[the remark this answers is not in the document]
Comment (anish-sahoo) [reply tools/test-harness/manifest.yaml:133]: depracted, we have replacement code with same functionality

File: tools/test-harness/manifest.yaml [hunk 7]
```diff
@@ -133,10 +141,61 @@ models:
...
+          name: "world"
+        expect:
+          type: contains
+          value: "hello"
+
+  - name: managed-weights
+    repo: local
+    base_dir: ../../examples
+    path: managed-weights
+    gpu: false
```
Comment (markphelps) [inline tools/test-harness/manifest.yaml:156]: This makes `managed-weights` part of the default CPU harness run, but that example depends on manually imported/git-ignored weights. A fresh contributor running the default manifest can fail before they have populated those weights. If we want to keep this in the manifest, can we gate it explicitly?

```suggestion
    gpu: false
    requires_env:
      - COG_MANAGED_WEIGHTS_READY
```

Alternatively, remove it from the default manifest or replace it with a lightweight managed-weights fixture using checked-in test data.

File: examples/hello-context/README.md [hunk 8]
```diff
@@ -0,0 +1,3 @@
+## hello-context
+
+A simple model that takes no inputs but will echo back any context provided with the prediction as the output.
```
Comment (markphelps) [inline examples/hello-context/README.md]: This README says the model takes no inputs, but `run.py` requires a `text` input and the harness supplies one. Can we make the README match the actual example?

```suggestion
A simple model that echoes the `text` input and any prediction context in the output.
```

File: examples/hello-concurrency/README.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+The key piece is the new `concurrency` field in the cog.yaml.
+
+```yaml
+concurrency:
+  max: 32
+```
+
+This combined with the async setup and predict methods in the predict.py allows Cog to run up to
+32 concurrent predictions. If cog reaches the max concurrency threshold it will reject subsequent
+predictions with a `409 Conflict` response.
```
Comment (markphelps) [inline examples/hello-concurrency/README.md:15]: This snippet/prose is stale after the migration: `cog.yaml` has `max: 4`, and the implementation is `run.py:Runner.run` rather than `predict.py`.

````suggestion
```yaml
concurrency:
  max: 4
```

This combined with the async setup and run methods in `run.py` allows Cog to run up to
4 concurrent predictions. If Cog reaches the max concurrency threshold it will reject subsequent
predictions with a `409 Conflict` response.
````

File: examples/hello-concurrency/README.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+### Telemetry
+
+It also uses the open-telemetry package to demonstrate how to collect telemetry for your model.
+
+This requires a file named `honeycomb_token.key` to be included in the image build.
+
+It will then start sending events to the `cog-model` data source. You can configure this by
+editing the `OTEL_SERVICE_NAME`. If you use a custom endpoint this can be configured via `OTEL_EXPORTER_OTLP_ENDPOINT`.
+
+Lastly, there is a section in predict.py that can be uncommented to run telemetry locally and print events to the console for debugging.
```
Comment (markphelps) [inline examples/hello-concurrency/README.md]: One more stale filename reference from the `predict.py` -> `run.py` migration.

```suggestion
Lastly, there is a section in `run.py` that can be uncommented to run telemetry locally and print events to the console for debugging.
```

**Review by markphelps — COMMENTED** [review]
Non-Secret review comments from a focused pass on the example migration and test harness. I left Secret/coglet scope comments out separately.

File: examples/experimental/resnet-managed-weights/README.md [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+Locally, cog assembles the weight files from the cache and bind-mounts them
+into the container at the `target` path. You don't need to push anything to test.
+
+### 3. Build and push
+
+```sh
+cog push
+```
+
+This builds the model image and pushes it to the registry specified by `image:`
```
Comment (markphelps) [inline examples/experimental/resnet-managed-weights/README.md]: This still refers to the old `image:` field, but this PR changed the example config to use `model:`.

```suggestion
This builds the model image and pushes it as the model named by `model:`
```

File: examples/experimental/resnet-managed-weights/README.md [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+build daemon on every `cog build`.
+
+## Adapting this for your own model
+
+1. Copy this directory as a starting point
+2. Edit `cog.yaml`:
+   - Change `source.uri` to your HuggingFace repo (or a local path)
+   - Adjust `exclude` patterns for the formats you don't need
+   - Set `target` to wherever your code expects to find the weights
+   - Set `image` to your registry destination (required for `cog push`)
```
Comment (markphelps) [inline examples/experimental/resnet-managed-weights/README.md]: Same stale `image:` guidance here; users following this would edit a field the example no longer uses.

```suggestion
   - Set `model` to your model name (required for `cog push`)
```

File: examples/resnet/README.md [hunk 13]
```diff
@@ -1,144 +1,20 @@
...
 
-ResNet50 image classifier (microsoft/resnet-50 from HuggingFace) packaged
-with v1 managed weights. Takes an image, returns top-3 ImageNet classes.
+This model tells you what's in an image. It's a good example of a deep
+learning model that's small enough to run without a GPU if you're demoing it.
 
-Use this as a starting point for packaging a real model with managed weights.
+It uses ResNet50 with the ImageNet weights that ship with torchvision, so
+there are no weight files to download or import -- torchvision fetches them
+the first time the model runs. Takes an image, returns the top-3 ImageNet
```
Comment (markphelps) [inline examples/resnet/README.md]: This description is internally inconsistent with the checked-in example: `cog.yaml` marks it GPU-enabled, and torchvision fetches/caches the ResNet checkpoint at setup rather than shipping it in the repo/image. Can we describe the actual runtime behavior?

```suggestion
This model tells you what's in an image. It's configured as a GPU example in `cog.yaml`.

It uses ResNet50 with ImageNet weights from torchvision. Torchvision fetches and caches the checkpoint the first time the model starts, so startup requires network access unless the checkpoint is already cached. Takes an image, returns the top-3 ImageNet
```

**Review by markphelps — CHANGES_REQUESTED** [review]
I think the coglet Secret input support should be pulled out into a separate PR. This PR is already large and is primarily about moving/modernizing examples; adding new runtime input coercion behavior changes the review surface and deserves its own focused tests/review.

For this PR, can we either:

1. remove the coglet Secret changes and any example/test-harness dependency on them, or
2. keep the hello-replicate example only if it works with existing released behavior / reads REPLICATE_API_TOKEN from env rather than requiring new coglet Secret coercion?

Then a follow-up PR can add Secret, Optional[Secret], and any list/edge-case behavior with dedicated coglet tests.

File: tools/test-harness/manifest.yaml (hunk 7 above)
Comment (anish-sahoo) [reply tools/test-harness/manifest.yaml:156]: Fixed — gated behind `requires_env: [COG_MANAGED_WEIGHTS_READY]` so it's skipped from the default harness run.

File: examples/hello-context/README.md (hunk 8 above)
Comment (anish-sahoo) [reply examples/hello-context/README.md]: Fixed — README now says it echoes the `text` input and any prediction context.

File: examples/hello-concurrency/README.md (hunk 9 above)
Comment (anish-sahoo) [reply examples/hello-concurrency/README.md:15]: Fixed — snippet updated to `max: 4` and the prose now refers to the async setup/run methods in `run.py`.

File: examples/hello-concurrency/README.md (hunk 10 above)
Comment (anish-sahoo) [reply examples/hello-concurrency/README.md]: Fixed — now references `run.py` instead of `predict.py`.

File: examples/experimental/resnet-managed-weights/README.md (hunk 11 above)
Comment (anish-sahoo) [reply examples/experimental/resnet-managed-weights/README.md]: Fixed — now says it pushes the model named by `model:`.

File: examples/experimental/resnet-managed-weights/README.md (hunk 12 above)
Comment (anish-sahoo) [reply examples/experimental/resnet-managed-weights/README.md]: Fixed — now instructs setting `model` to your model name.

File: examples/resnet/README.md (hunk 13 above)
Comment (anish-sahoo) [reply examples/resnet/README.md]: Fixed — README now states it's a GPU example and that torchvision fetches/caches the checkpoint at startup (network needed unless cached).

**Comment by anish-sahoo** [comment]
Re: pulling the coglet `Secret` support into its own PR — done. The coglet Secret coercion (`Secret`, `Optional[Secret]`, `Secret | None`, with `list[Secret]` excluded) plus dedicated classification tests and the `hello-replicate` example now live in #3057. This PR drops the coglet Secret changes, the `hello-replicate` example + its harness entry, and the `Secret` usage in the `scalar-types` fixture, so it's back to being just the example migration + test-harness work.

File: tools/test-harness/manifest.yaml [hunk 14]
```diff
@@ -37,7 +41,8 @@ models:
   # ── cog-examples (CPU) ──────────────────────────────────────────────
 
   - name: hello-world
-    repo: replicate/cog-examples
+    repo: local
+    base_dir: ../../examples
```
Comment (markphelps) [inline tools/test-harness/manifest.yaml]: nit: seems still to have to specify `repo` + `base_dir`. to me `base_dir` implies local, so it should check for base `dir` first and assume local if repo is not set

**Review by markphelps — COMMENTED** [review]
one nit

File: tools/test-harness/manifest.yaml (hunk 14 above)
Comment (anish-sahoo) [reply tools/test-harness/manifest.yaml]: updated code to remove `repo: local` and instead support `base_dir`

**Review by markphelps — APPROVED** [review]
looks great! thank you for taking this on
